extensions/cornerstone/src/init.js

```diff
@@ -102,6 +102,25 @@ export default function init({ servicesManager, configuration }) {
     csTools.BrushTool,
   ];
 
+  /* Measurement Service Events */
+  cornerstone.events.addEventListener(
+    cornerstone.EVENTS.ELEMENT_ENABLED,
+    event => {
+      MeasurementService.subscribe(
+        MeasurementService.getEvents().MEASUREMENT_ADDED,
+        measurement =>
+          console.log('[subscribe] Measurement added or updated', measurement)
+      );
+      event.detail.element.addEventListener(
+        csTools.EVENTS.MEASUREMENT_ADDED,
+        event => {
+          console.log('[addOrUpdate] Adding new measurement...', event.detail);
+          MeasurementService.addOrUpdate({ id: 1, data: event.detail });
```

Comment:
This is where we need James' mapping magic before passing data. We've hit on the rough bits that need to be there.

Reply:
We may add a helper to the MeasurementService that allows you to grab any measurement in 3D or 2D coords. That is where we would use James' helper.

Or, if we don't want to mix the internal representation, we could store all points in xyz, but would need to key off sopInstanceUid being set to know that the original was 2D